File: app/src/main/java/com/yawar/memo/modelView/RequestCallViewModel.java
package com.yawar.memo.modelView;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.UUID;

public class RequestCallViewModel extends ViewModel {
//    RequestCallRepo  requestCallRepo = BaseApp.getInstance().getRequestCallRepo();

    public MutableLiveData<String> peerIdRecived;
    //    private   Boolean _isVideoForMe;
    public MutableLiveData<Boolean> isVideoForMe;
//    private   Boolean _isVideoForYou;
    public MutableLiveData<Boolean> isVideoForYou;
//    private   Boolean _endCall;
    public MutableLiveData<Boolean> endCall;
//    private   Boolean _rining;
    public MutableLiveData<String> rining;

    public MutableLiveData<Boolean> isAudio;
    public MutableLiveData<Boolean> connected;

    private int time = 0;



    public MutableLiveData<Boolean> isSpeaker;

    private String callId;



    public RequestCallViewModel() {
        this.isVideoForMe = new MutableLiveData<>(false);
        this.isVideoForYou=new MutableLiveData<>(false);
        this.endCall = new MutableLiveData<>(false);
        this.rining = new MutableLiveData<>("connect");
        this.peerIdRecived = new MutableLiveData<>("no connect");
        this.isAudio = new MutableLiveData<>(true);
        this.isSpeaker = new MutableLiveData<>(true);
        this.connected = new MutableLiveData<>(false);


    }

    public MutableLiveData<Boolean> getIsVideoForMe() {
        return this.isVideoForMe;
    }

    public void setIsVideoForMe(boolean isVideoForMe) {
        this.isVideoForMe.setValue(isVideoForMe);
    }

    public MutableLiveData<Boolean> getIsVideoForYou() {
        return this.isVideoForYou;
    }

    public void setIsVideoForYou( boolean isVideoForYou) {
        this.isVideoForYou.setValue(isVideoForYou); ;
    }

    public MutableLiveData<Boolean> getEndCall() {
        return this.endCall;
    }

    public void setEndCall(boolean endCall) {
        this.endCall.setValue( endCall);
    }

    public MutableLiveData<String> getRining() {
        return this.rining;
    }

    public void setRining(String rining) {
        this.rining.setValue(rining);
    }

    public MutableLiveData<Boolean> getIsSpeaker() {
        return this.isSpeaker;
    }

    public void setIsSpeaker(boolean isSpeaker) {
        this.isSpeaker.setValue(isSpeaker);
    }
    public void setPeerId(String peerId) {
        this.peerIdRecived.setValue(peerId);
    }
    public MutableLiveData<String> getPeerId( ) {
       return this.peerIdRecived;
    }
    public void setAudio(Boolean audio) {
        this.isAudio.setValue(audio);
    }
    public MutableLiveData<Boolean> getAudio( ) {
        return this.isAudio;
    }
    public String getCallId( ) {
        return this.callId;
    }


    public void setConnected(Boolean connected) {
        this.connected.postValue(connected);
    }
    public MutableLiveData<Boolean> getConnected( ) {
        return this.connected;
    }

    public void setCallId() {

        this.callId = UUID.randomUUID().toString();
    }
   public  void setTime (int time){
        this.time = time;
   }

    public int getTime() {
        return time;
    }

    public  String getTimeString(){
       int seconds = this.time % 60;
       int minutes = this.time / 60;
       int hour = minutes/60;
       String stringTime = String.format("%02d:%02d:%02d",hour, minutes, seconds);
       return  stringTime;
   }

}

File: app/src/main/java/com/yawar/memo/ui/requestCall/RequestCallViewModel.kt
package com.yawar.memo.ui.requestCall

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import java.util.*

class RequestCallViewModel : ViewModel() {
    //    RequestCallRepo  requestCallRepo = BaseApp.getInstance().getRequestCallRepo();
    private val _peerIdRecived =  MutableLiveData<String>("no connect")
    val peerIdRecived : LiveData<String>
        get() = _peerIdRecived

    private val _rining =  MutableLiveData<String>("connect")
    val rining : LiveData<String>
        get() = _rining

    private val _isVideoForMe =  MutableLiveData<Boolean>(false)
    val isVideoForMe : LiveData<Boolean>
        get() = _isVideoForMe

    private val _isVideoForYou =  MutableLiveData<Boolean>(false)
    val isVideoForYou : LiveData<Boolean>
        get() = _isVideoForYou

    private val _isAudio =  MutableLiveData<Boolean>(true)
    val isAudio : LiveData<Boolean>
        get() = _isAudio

    private val _endCall =  MutableLiveData<Boolean>(false)
    val endCall : LiveData<Boolean>
        get() = _endCall

    private val _isSpeaker =  MutableLiveData<Boolean>(true)
    val isSpeaker : LiveData<Boolean>
        get() = _isSpeaker

    private val _connected =  MutableLiveData<Boolean>(false)
    val connected : LiveData<Boolean>
        get() = _connected

    private var time = 0


    private var callId: String? = null


//    fun getIsVideoForMe(): LiveData<Boolean> {
//        return _isVideoForMe
//    }

    fun setIsVideoForMe(isVideoForMe: Boolean) {
        _isVideoForMe.value = isVideoForMe
    }


    fun setIsVideoForYou(isVideoForYou: Boolean) {
        _isVideoForYou.value = isVideoForYou
    }

    fun setEndCalll(endCall: Boolean) {
        _endCall.value = endCall
    }


    fun setRining(rining: String) {
        _rining.value = rining
    }



    fun setIsSpeaker(isSpeaker: Boolean) {
        _isSpeaker.value  = isSpeaker
    }

    fun setPeerId(peerId: String) {
        _peerIdRecived.value = peerId
    }



    fun setAudio(audio: Boolean) {
        _isAudio.value = audio
    }



    fun getCallId(): String? {
        return callId
    }

    fun setConnected(connected: Boolean) {
        _connected.postValue(connected)
    }



    fun setCallId() {
        callId = UUID.randomUUID().toString()
    }

    fun setTime(_time: Int) {
        time = _time
    }

    fun getTime(): Int {
        return time
    }
    fun getTimeString(): String {
        val seconds = time % 60
        val minutes = time / 60
        val hour = minutes / 60
        return String.format("%02d:%02d:%02d", hour, minutes, seconds)
    }


}